libs/utils/android/workloads/gmaps.py

```diff
@@ -0,0 +1,123 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2015, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import re
+import os
+import logging
+
+from time import sleep
+
+from android import Screen, System
+from android.workload import Workload
+
+
+class GMaps(Workload):
+    """
+    Android GMaps workload
+    """
+
+    # Package required by this workload
+    package = 'com.google.android.apps.maps'
+    action = 'android.intent.action.VIEW'
+
+    def __init__(self, test_env):
+        super(GMaps, self).__init__(test_env)
+        self._log = logging.getLogger('GMaps')
+        self._log.debug('Workload created')
+
+        # Set of output data reported by Jankbench
+        self.db_file = None
+
+    def run(self, out_dir, location_search, swipe_count=5, collect=''):
+        """
+        Run single Gmaps workload.
+
+        :param out_dir: Path to experiment directory where to store results.
+        :type out_dir: str
+
+        :param location_search: Search string to be used in GMaps
+        :type video_url: str
+
+        :param swipe_count: Number of sets of (left, right, up, down) swipes to do
+        :type video_duration_s: int
+
+        :param collect: Specifies what to collect. Possible values:
+            - 'energy'
+            - 'systrace'
+            - 'ftrace'
+            - any combination of the above
+        :type collect: list(str)
+        """
+
+        # Keep track of mandatory parameters
+        self.out_dir = out_dir
+        self.collect = collect
+
+        # Unlock device screen (assume no password required)
+        System.menu(self._target)
+        # Press Back button to be sure we run the video from the start
+        System.back(self._target)
+
```

Comment:
We also set here the brightness to 0 for the display to not have too much impact on the energy measurements:

```
        # Set min brightness
        Screen.set_brightness(self._target, auto=False, percent=0)
```

and restore it to auto at the end:

```
        # Set brightness back to auto
        Screen.set_brightness(self._target, auto=True)
```